interview-questions.md

```diff
@@ -7,60 +7,60 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Researched answer:
+  Researched answer: According to the syllabus "functions are a fundamental building block in JavaScript and in programming in general. Functions are reusable pieces of code that only execute when invoked and always return an output." By reading that it lets me know that I was on the right track with my answer. 
 
 
 
 2. In regards to functions, what is the difference between an argument and a parameter?
 
-  Your answer:
+  Your answer: A parameter is an external source of information why the arguments process that parameter for a outcome For instants if I was designing a function for a Thermometer that told you if the temperature was above or below the boiling point. I would first create a parameter called boiling point then create the Arguments that if the temperature was below 212° it would display and tell you so if it was higher and it will tell you that if you were right on the money it would tell you that you were at the boiling point.
 
-  Researched answer:
+  Researched answer: Doing the research I found that I'm on the right track but the way that my answer explains it the function wouldn't be called boiling point it would be called temp where you could put in the temperatures so that the code would run in order to know if your above, below or right on the money.
 
 
 
 3. What is an index? What is the difference between index and value?
 
-  Your answer:
+  Your answer: An index is a way to identify elements and what place they hold.So therefore the difference between a value and an index is that the index is a placeholder starting at zero and counting up and a value is the actual value of what you've put in, like data type number, string And so forth.
 
-  Researched answer:
+  Researched answer: According to my research of value is part of an array. The array can hold as many values Do you want and there are certain commands that you can use to change the values in an array. Now an index is a way to find out how many values there are in that array and what number order they're in there is such a thing called zero indexing.
 
 
 
 4. What is a built-in method?
 
-  Your answer:
+  Your answer: These are methods that are built into the program to help us code quicker and easier. They are tools for developers to use such as a mechanic uses a wrench we use built-in methods. Could you imagine if we didn't have them we would have to re-create the same methods over and over and over again. It's like asking a mechanic to do anything without using a wrench. It's not that it's impossible it would just take a lot longer to accomplish whatever you were trying to do.
 
-  Researched answer:
+  Researched answer: There are many built in methods in arrays, Because arrays sotre alot of date And the built-in methods make it easy for us to manipulate that data. We used a lot of them in our Assessments this week. A built-in method is a small piece of code that has been added to accomplish a common task. 
```

Comment:
yeah and there are methods for: Strings, Arrays, Numbers, Objects, and so on

Great Answer!